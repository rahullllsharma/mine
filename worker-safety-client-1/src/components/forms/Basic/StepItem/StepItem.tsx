import type { IconName } from "@urbint/silica";
import { Icon } from "@urbint/silica";
import classnames from "classnames";

export type Status =
  | "default"
  | "current"
  | "saved"
  | "saved_current"
  | "error";

type StepItemProps = {
  status?: Status;
  truncate?: boolean;
  label: string;
  onClick: () => void;
};

type StatusIconClassName = {
  [key in Status]: {
    icon: IconName;
    iconClassName?: string;
    buttonClassName?: string;
  };
};

const statusIconClassNames: StatusIconClassName = {
  default: {
    icon: "circle",
    iconClassName: "text-neutral-shade-58",
    buttonClassName: "border-2 border-transparent box-border",
  },
  current: {
    icon: "circle",
    iconClassName: "text-neutral-shade-58",
    buttonClassName: "border-2 border-neutral-shade-100 box-border",
  },
  saved: {
    icon: "circle_check",
    iconClassName: "text-brand-urbint-40",
    buttonClassName: "border-2 border-transparent bg-brand-urbint-10",
  },
  saved_current: {
    icon: "circle_check",
    iconClassName: "text-brand-urbint-40",
    buttonClassName:
      "border-2 bg-brand-urbint-10 border-brand-urbint-30 box-border",
  },
  error: {
    icon: "error",
    iconClassName: "text-system-error-40",
    buttonClassName:
      "border-2 border-system-error-40 bg-system-error-10 box-border",
  },
};

const StepItem = ({
  label,
  status = "default",
  truncate = false,
  onClick,
}: StepItemProps) => {
  const { icon, iconClassName, buttonClassName } = statusIconClassNames[status];

  return (
    <button
      className={classnames(
        "p-3 bg-neutral-shade-3 rounded flex w-full items-center text-left",
        buttonClassName
      )}
      onClick={onClick}
    >
      <Icon name={icon} className={iconClassName} />
      <span
        className={classnames(
          "ml-2 text-base font-normal text-neutral-shade-100",
          { ["truncate"]: truncate }
        )}
      >
        {label}
      </span>
    </button>
  );
};

export { StepItem };
