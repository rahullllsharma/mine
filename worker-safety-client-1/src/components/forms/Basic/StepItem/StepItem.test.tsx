import { fireEvent, render, screen } from "@testing-library/react";

import { StepItem } from "./StepItem";

describe(StepItem.name, () => {
  it("renders correctly default step", () => {
    const { asFragment } = render(
      <StepItem label="Test" onClick={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly current step", () => {
    const { asFragment } = render(
      <StepItem status="current" label="Test" onClick={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly visited step", () => {
    const { asFragment } = render(
      <StepItem status="saved" label="Test" onClick={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe("user actions", () => {
    it("behaves correctly when user clicks", () => {
      const onClick = jest.fn();
      render(<StepItem label="Test" onClick={onClick} />);

      fireEvent.click(screen.getByRole("button"));

      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
