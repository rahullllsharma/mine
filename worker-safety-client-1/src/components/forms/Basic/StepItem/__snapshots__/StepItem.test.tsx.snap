// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepItem renders correctly current step 1`] = `
<DocumentFragment>
  <button
    class="p-3 bg-neutral-shade-3 rounded flex w-full items-center text-left border-2 border-neutral-shade-100 box-border"
  >
    <i
      aria-hidden="true"
      class="text-neutral-shade-58"
    >
      <svg
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          fill="currentColor"
          r="10"
        />
      </svg>
    </i>
    <span
      class="ml-2 text-base font-normal text-neutral-shade-100"
    >
      Test
    </span>
  </button>
</DocumentFragment>
`;

exports[`StepItem renders correctly default step 1`] = `
<DocumentFragment>
  <button
    class="p-3 bg-neutral-shade-3 rounded flex w-full items-center text-left border-2 border-transparent box-border"
  >
    <i
      aria-hidden="true"
      class="text-neutral-shade-58"
    >
      <svg
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          fill="currentColor"
          r="10"
        />
      </svg>
    </i>
    <span
      class="ml-2 text-base font-normal text-neutral-shade-100"
    >
      Test
    </span>
  </button>
</DocumentFragment>
`;

exports[`StepItem renders correctly visited step 1`] = `
<DocumentFragment>
  <button
    class="p-3 bg-neutral-shade-3 rounded flex w-full items-center text-left border-2 border-transparent bg-brand-urbint-10"
  >
    <i
      aria-hidden="true"
      class="ci-circle_check text-brand-urbint-40"
    />
    <span
      class="ml-2 text-base font-normal text-neutral-shade-100"
    >
      Test
    </span>
  </button>
</DocumentFragment>
`;
